plugin/storage/badger/factory.go

```diff
@@ -164,6 +167,11 @@ func (f *Factory) CreateDependencyReader() (dependencystore.Reader, error) {
 	return depStore.NewDependencyStore(sr), nil
 }
 
+// CreateLockAndSamplingStore implements storage.ArchiveFactory. It is not supported by this storage plugin.
+func (f *Factory) CreateLockAndSamplingStore() (distributedlock.Lock, samplingstore.Store, error) {
```

Comment:
Another approach we could take that would minimize changes in this diff like this no-op addition to all factories is to just create a new optional interface SamplingStoreFactory and typecast-check for it where needed. I don't like all this noise being added to factories that don't need it.

Reply:
agreed. done.